tests/tests.rs

```diff
@@ -500,8 +500,12 @@ fn result_match() {
 
     match ok() {
         Ok(()) => {},
-        Err(Error(ErrorKind::Msg(_), _)) => {},
-        Err(..) => {},
+        Err(e) => {
+            match *e.kind() {
+                ErrorKind::Msg(_) => {}
+                _ => {}
+            }
+        }
```

Comment:
I'm not very satisfied with this new behaviour, it's less straightforward. Not sure how to improve it though...

Reply:
box_patterns will be stabilized some day, no?

Reply:
@Yamakaky do you have a preferred method of how we might land this? I'd personally like to see this as at least an option, it's what I'd want for libraries 99% of the time I believe.